pkg/virt-controller/services/template.go

```diff
@@ -961,9 +962,15 @@ func getNamespaceAndNetworkName(vmi *v1.VirtualMachineInstance, fullNetworkName
 
 func getNetworkToResourceMap(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) (networkToResourceMap map[string]string, err error) {
 	networkToResourceMap = make(map[string]string)
-	for _, network := range vmi.Spec.Networks {
+	for idx, network := range vmi.Spec.Networks {
 		if network.Multus != nil {
-			namespace, networkName := getNamespaceAndNetworkName(vmi, network.Multus.NetworkName)
+			networkSource := v1.NetworkSource{Npwgv1: network.Multus}
+			npwgv1Network := v1.Network{Name: network.Name, NetworkSource: networkSource}
+			vmi.Spec.Networks[idx] = npwgv1Network
+			network.Npwgv1 = network.Multus
```

Comment:
what does this line do? AFAIU you've just overridden the network in vmi.Spec to a network of the new format, so this network is no longer attached to the list. Why do you need to modify it instead of just leaving it to garbage collector since it's, presumably, no longer attached to anywhere?

Reply:
In the next line we check for network.Npwgv1, hence network.Npwgv1 has to be set:
https://github.com/kubevirt/kubevirt/blob/1983659767f9737533edd710e23588bf8e8782a8/pkg/virt-controller/services/template.go#L970-L972